### PR TITLE
Registrationemail

### PR DESCRIPTION
**Pull request by zspadar**

### CODE REVIEW

File: src/main/java/com/greenfox/notification/controller/HeartbeatController.java [hunk 1]
```diff
@@ -30,4 +43,25 @@ public Object getHeartbeats(HttpServletRequest request) throws Exception {
     rabbitMQ.consume("heartbeat");
     return responseService.checkForResponse();
   }
+
+  @PostMapping("/email/registration")
```
Comment (EggDice) [inline src/main/java/com/greenfox/notification/controller/HeartbeatController.java]: Move the endpoint to a separate file

File: src/main/java/com/greenfox/notification/controller/HeartbeatController.java [hunk 2]
```diff
@@ -30,4 +43,25 @@ public Object getHeartbeats(HttpServletRequest request) throws Exception {
...
+  public Data registration(@RequestBody Data data) throws IOException {
+    Email from = new Email("test@example.com");
+    String subject = "I'm replacing the subject tag";
+    Email to = new Email(data.getAttributes().getEmail());
+    com.sendgrid.Content content = new com.sendgrid.Content("text/plain", "vilmoskorte");
+    Mail mail = new Mail(from, subject, to, content);
+    mail.personalization.get(0).addSubstitution("-name-", data.getAttributes().getName());
+    mail.setTemplateId("0783a75f-044a-4725-8624-425cc41b7735");
+    SendGrid sg = new SendGrid(System.getenv("SENDGRID_API_KEY"));
```
Comment (EggDice) [inline src/main/java/com/greenfox/notification/controller/HeartbeatController.java]: Create a service for sending emails

File: src/main/java/com/greenfox/notification/controller/HeartbeatController.java [hunk 3]
```diff
@@ -30,4 +43,25 @@ public Object getHeartbeats(HttpServletRequest request) throws Exception {
...
+    Request request = new Request();
+    request.setMethod(Method.POST);
+    request.setEndpoint("mail/send");
+    request.setBody(mail.build());
+    Response response = sg.api(request);
+    System.out.println(response.getStatusCode());
+    System.out.println(response.getBody());
+    System.out.println(response.getHeaders());
```
Comment (EggDice) [inline src/main/java/com/greenfox/notification/controller/HeartbeatController.java]: Don't leave `System.out.println` in the code, use the logger

Comment (zspadar) [reply src/main/java/com/greenfox/notification/controller/HeartbeatController.java]: @EggDice 
Please, see the modified version.
Thanks,
Team Emerald

File: src/main/java/com/greenfox/notification/controller/ConfirmationController.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import com.greenfox.notification.service.EmailSenderService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.servlet.http.HttpServletRequest;
+
+@RestController
+public class ConfirmationController {
```
Comment (zspadar) [inline src/main/java/com/greenfox/notification/controller/ConfirmationController.java]: Include somehow its about the registration notification

File: src/main/java/com/greenfox/notification/model/interfaces/MessageQueue.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+public interface MessageQueue<T, U> {
+  void push(HttpServletRequest request, T queue, U message) throws IOException;
+
+  //Closeable consume(Function<T, Void> processor);
+
+  //T consumeNext();
```
Comment (zspadar) [inline src/main/java/com/greenfox/notification/model/interfaces/MessageQueue.java:11]: don't leave code in comments

File: src/main/java/com/greenfox/notification/service/EmailGenerator.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+public class EmailGenerator {
+  private Email sender;
+  private String subject;
+
+  public EmailGenerator(){
+    this.subject = "Registration process";
+    this.sender = new Email("test@example.com");
```
Comment (zspadar) [inline src/main/java/com/greenfox/notification/service/EmailGenerator.java]: it should come from config

File: src/main/java/com/greenfox/notification/service/EmailSenderService.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+import com.greenfox.notification.model.classes.Data;
+import com.sendgrid.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.servlet.http.HttpServletRequest;
+
+@Service
+public class EmailSenderService {
```
Comment (zspadar) [inline src/main/java/com/greenfox/notification/service/EmailSenderService.java:11]: Separate this service to:
 - Something that push it to the queue
 - Something that consume it from the queue
 - Something that sends an email

**Pull request merged**